src/uct/cuda/cuda_ipc/cuda_ipc_md.c

```diff
@@ -105,11 +105,69 @@ uct_cuda_ipc_md_query(uct_md_h md, uct_md_attr_v2_t *md_attr)
     return UCS_OK;
 }
 
+static ucs_status_t
+uct_cuda_ipc_mem_add_reg_push_ctx(CUdeviceptr address, CUdevice *cuda_device_p)
+{
+    CUdevice cuda_device = CU_DEVICE_INVALID;
+    #define UCT_CUDA_IPC_NUM_ATTRS 2
+    CUpointer_attribute attr_type[UCT_CUDA_IPC_NUM_ATTRS];
+    void *attr_data[UCT_CUDA_IPC_NUM_ATTRS];
+    CUcontext cuda_ctx;
+    int cuda_device_ordinal;
+    ucs_status_t status;
+
+    attr_type[0] = CU_POINTER_ATTRIBUTE_CONTEXT;
+    attr_data[0] = &cuda_ctx;
+    attr_type[1] = CU_POINTER_ATTRIBUTE_DEVICE_ORDINAL;
+    attr_data[1] = &cuda_device_ordinal;
+    
+    status = UCT_CUDADRV_FUNC_LOG_ERR(
+            cuPointerGetAttributes(UCT_CUDA_IPC_NUM_ATTRS, attr_type, attr_data,
+                                   address));
+    if (status != UCS_OK) {
+        return status;
+    }
```

Comment:
can we use active context (if such exists), even if the buffer was allocated on another context?

Reply:
for next PR, can also unite with uct_cuda_copy_md_sync_memops_get_address_range